### ISSUE TITLE
Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'".

### ISSUE THREAD

**Issue by Arcanemagus** [AtomLinter/linter-stylelint#90]
This appears to be an issue with the `function-bind` dependency of the `tape` dependency of `stylelint`, I'm just not sure if this is an Atom issue, a `stylelint` issue, or a `function-bind` issue :confused: 

[Enter steps to reproduce below:]

1. Install / Load `linter-stylelint`

**Atom Version**: 1.6.0-beta3
**System**: Microsoft Windows 10 Pro Insider Preview
**Thrown From**: [linter-stylelint](https://github.com/AtomLinter/linter-stylelint) package, v1.11.0

### Stack Trace

Failed to load the linter-stylelint package

```
At Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'".


EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'".

    at Function (native)
    at Function.bind (C:\Users\Landon\.atom\packages\linter-stylelint\node_modules\stylelint\node_modules\tape\node_modules\function-bind\index.js:37:17)
    at Object.<anonymous> (C:\Users\Landon\.atom\packages\linter-stylelint\node_modules\stylelint\node_modules\tape\node_modules\has\src\index.js:3:23)
    at Module._compile (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .js] (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\compile-cache.js:208:21)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\native-compile-cache.js:50:27)
    at Object.<anonymous> (C:\Users\Landon\.atom\packages\linter-stylelint\node_modules\stylelint\node_modules\tape\lib\test.js:6:11)
    at Module._compile (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .js] (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\compile-cache.js:208:21)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\native-compile-cache.js:50:27)
    at Object.<anonymous> (C:\Users\Landon\.atom\packages\linter-stylelint\node_modules\stylelint\node_modules\tape\index.js:3:12)
    at Module._compile (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .js] (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\compile-cache.js:208:21)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\native-compile-cache.js:50:27)
    at Object.<anonymous> (C:\Users\Landon\.atom\packages\linter-stylelint\node_modules\stylelint\dist\testUtils\ruleTester.js:130:13)
    at Module._compile (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .js] (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\compile-cache.js:208:21)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\native-compile-cache.js:50:27)
    at Object.<anonymous> (C:\Users\Landon\.atom\packages\linter-stylelint\node_modules\stylelint\dist\index.js:21:19)
    at Module._compile (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .js] (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\compile-cache.js:208:21)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\native-compile-cache.js:50:27)
    at Object.<anonymous> (C:/Users/Landon/.atom/packages/linter-stylelint/lib/index.js:3:17)
    at Module._compile (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .js] (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\compile-cache.js:208:21)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\native-compile-cache.js:50:27)
    at Package.module.exports.Package.requireMainModule (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\package.js:720:27)
    at C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\package.js:116:28
    at Package.module.exports.Package.measure (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\package.js:92:15)
    at Package.module.exports.Package.load (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\package.js:106:12)
    at PackageManager.module.exports.PackageManager.loadPackage (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\package-manager.js:445:14)
    at PackageManager.module.exports.PackageManager.activatePackage (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\package-manager.js:524:30)
    at C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\node_modules\settings-view\lib\package-manager.js:444:29
    at exit (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\node_modules\settings-view\lib\package-manager.js:73:16)
    at triggerExitCallback (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\buffered-process.js:213:47)
    at C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\buffered-process.js:227:18
    at Socket.<anonymous> (C:\Users\Landon\AppData\Local\atom\app-1.6.0-beta3\resources\app.asar\src\buffered-process.js:98:18)
    at emitOne (events.js:82:20)
    at Socket.emit (events.js:169:7)
    at Pipe._onclose (net.js:469:12)
```

### Commands

```
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "language-gfm"
    ],
    "themes": [
      "seti-ui",
      "seti-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
atom-autocomplete-php, v0.18.7
autocomplete-python, v1.6.1
editorconfig, v1.2.4
file-watcher, v0.4.0
highlight-selected, v0.11.2
language-chef, v0.8.0
language-ini, v1.14.0
language-markdown, v0.10.1
language-powershell, v2.1.0
language-r, v0.4.0
language-reg, v0.0.0
linter, v1.11.3
linter-coffeelint, v1.1.2
linter-csslint, v1.3.0
linter-eslint, v6.0.0
linter-flake8, v1.10.1
linter-js-yaml, v1.2.5
linter-jscs, v3.4.6
linter-jshint, v1.3.1
linter-markdown, v1.4.2
linter-perl, v0.7.2
linter-php, v1.1.8
linter-phpcs, v1.3.0
linter-phpmd, v1.4.1
linter-pylint, v1.2.0
linter-reek, v2.1.3
linter-rubocop, v0.4.7
linter-ruby, v1.2.1
linter-stylelint, v1.11.0
minimap, v4.19.0
minimap-find-and-replace, v4.4.0
minimap-git-diff, v4.1.8
minimap-highlight-selected, v4.3.1
minimap-linter, v1.1.1
minimap-selection, v4.3.1
pigments, v0.23.1
seti-syntax, v0.4.1
seti-ui, v0.8.1
svn, v0.0.9
tab-control, v0.6.9
trailing-spaces, v0.3.2

# Dev
No dev packages
```

**Comment by Arcanemagus** [comment]
@davidtheclark The changes that broke this were introduced in `stylint@4.3.2` (4.3.0 and 4.3.1 were broken for other reasons already fixed there).

Is this an issue with the Atom environment, or your dependency? Do you want an issue filed over there?

**Comment by steelbrain** [comment]
Looks like stylint is doing `eval` somewhere which is disallowed by Atom, either we can use Nath's `loophole` or execute it in a worker or child process

**Comment by Arcanemagus** [comment]
@steelbrain The stack trace goes down to [this](https://github.com/Raynos/function-bind/blob/master/index.js), which is rather suspicious looking, and seems the equivalent of `eval()` to me.

**Comment by steelbrain** [comment]
This is the `eval()` here https://github.com/Raynos/function-bind/blob/dd96c56720034a3c1ffee10b8a59a6f7c53e24ad/index.js#L38

**Comment by steelbrain** [comment]
Here is the culprit line https://github.com/substack/tape/blob/87e4bec19bac80402011347f89b3647425d09c5e/lib/results.js#L8

**Comment by davidtheclark** [comment]
Ugh. Maybe I will take extract stylelint's `ruleTester` again. It was pulled into the main repo in 4.3: that's why the tape dependency was introduced, which is why we have this mess.

**Comment by steelbrain** [comment]
Opened this bug in tape https://github.com/substack/tape/issues/253

**Comment by davidtheclark** [comment]
I have backed out of this change, so 4.3.3 should work just fine (no tape dependency).

**Comment by Arcanemagus** [comment]
@davidtheclark thanks for the super fast response! I'll push out a new version in a few minutes depending on `^4.3.3` :wink:.

**Comment by Arcanemagus** [comment]
v1.11.2 published depending on `stylelint@^4.3.3`.

**Comment by steelbrain** [comment]
@davidtheclark @ljharb Just published a `function-bind` release that works on Atom, so you can use the `tape` dependency again!
